openfl-workspace/torch_cnn_mnist/README.md

```diff
@@ -0,0 +1,212 @@
+## Instantiating a Workspace from Torch Template
+To instantiate a workspace from the torch_cnn_mnist template, you can use the fx workspace create command. This allows you to quickly set up a new workspace based on a predefined configuration and template.
+1. Ensure the necessary dependencies are installed.
+"""
+pip install virtualenv
+mkdir ~/openfl-quickstart
+virtualenv ~/openfl-quickstart/venv
+source ~/openfl-quickstart/venv/bin/activate
+pip install openfl
+"""
+2. Creating the Workspace Folder
+
+"""
+cd ~/openfl-quickstart
+fx workspace create --template torch_template --prefix fl_workspace
+cd ~/openfl-quickstart/fl_workspace
+"""
+
+Here’s a step-by-step guide on how to do this:
+
+## Directory Structure
+The taskrunner workspace has the following file structure:
+```
+taskrunner
+├── requirements.txt      # defines the required software packages
+└── plan
+    ├── plan.yaml         # the Federated Learning plan declaration
+    ├── cols.yaml         # holds the list of authorized collaborators
+    ├── data.yaml         # holds the collaborator data set path
+    ├── defaults          # path to the default values for the FL plan
+├── src
+    ├── __init__.py       # treat src as a Python package
+    └── cnn_model.py      # centralized CNN model, ready for use in federated learning
+    ├── dataloader.py     # data loader module
+    └── taskrunner.py     # task runner module
+```
+
+## Directory Breakdown:
+* requirements.txt: This file lists all the Python dependencies required to run the TaskRunner API and its components. Ensure you install these dependencies by running pip install -r requirements.txt.
+* plan: Contains configuration files for federated learning:
+    - plan.yaml: The main Federated Learning plan declaration, defining the structure of the federated learning workflow.
+    - cols.yaml: A list of authorized collaborators for the federated learning task.
+    - data.yaml: Specifies the path to the data set for each collaborator.
+    - defaults: Path to the default configuration values for the federated learning plan.
+* src: This directory contains the Python modules used for federated learning:
+    - init.py: Marks the src directory as a Python package, allowing you to import modules within the directory.
+    - cnn_model.py: Defines the Convolutional Neural Network (CNN) model for federated learning.
+    - dataloader.py: A module responsible for loading and processing datasets for the federated learning task.
+    - taskrunner.py: The core task runner module that manages the execution of federated learning tasks.
+
+## Defining the Data Loader
+The data loader in OpenFL is responsible for batching and iterating through the dataset that will be used for local training and validation on each collaborator node. The PyTorchMNISTInMemory class is designed to handle the MNIST dataset, ensuring it is properly loaded and preprocessed for federated learning experiments.
```

Comment:
-> The PyTorchMNISTInMemory class is designed to load and shard the MNIST data set "on the fly".

Reply:
resolved in be37d1c1e35fee2cce4ee8907547c861e28e6d8f